src/endpoint/s3/ops/s3_get_object.js

```diff
@@ -100,20 +101,12 @@ async function get_object(req, res) {
         }
     }
 
-    let read_stream;
-
-    // on http disconnection close the read stream to stop from buffering more data
-    req.on('aborted', () => {
-        dbg.log0('request aborted:', req.path);
-        if (read_stream && read_stream.close) read_stream.close();
-    });
-    res.on('error', err => {
-        dbg.log0('response error:', err, req.path);
-        if (read_stream && read_stream.close) read_stream.close();
-    });
-
-    read_stream = await req.object_sdk.read_object_stream(params, res);
+    const read_stream = await req.object_sdk.read_object_stream(params, res);
     if (read_stream) {
+        // on http disconnection close the read stream to stop from buffering more data
+        if (read_stream.close) {
+            req.object_sdk.abort_controller.signal.onabort = () => read_stream.close();
```

Comment:
Don't override the onabort function, instead add an event handler to allow others to listen for it as well -
```suggestion
            req.object_sdk.abort_controller.signal.once('abort', () => read_stream.close());
```

Reply:
Added the event_listener as we discussed on DM